frontend/javascripts/oxalis/view/right-border-tabs/segments_tab/segments_view.tsx

```diff
@@ -1088,39 +1241,82 @@ class SegmentsView extends React.Component<Props, State> {
     });
   };
 
-  handleChangeMeshVisibilityInGroup = (layerName: string, groupId: number, isVisible: boolean) => {
+  handleChangeMeshVisibilityInGroup = (
+    layerName: string,
+    groupId: number | null,
+    isVisible: boolean,
+  ) => {
     this.handlePerSegment(groupId, (segment) => {
       if (Store.getState().localSegmentationData[layerName].isosurfaces[segment.id] != null) {
         Store.dispatch(updateIsosurfaceVisibilityAction(layerName, segment.id, isVisible));
       }
     });
   };
 
-  handleLoadMeshesAdHoc = (groupId: number) => {
+  handleLoadMeshesAdHoc = (groupId: number | null) => {
     this.handlePerSegment(groupId, (segment) => {
       if (segment.somePosition == null) return;
       this.props.loadAdHocMesh(segment.id, segment.somePosition);
     });
   };
 
-  getSegmentsWithMissingLocation = (groupId: number): number[] => {
-    const segmentGroup = this.getSegmentsOfGroup(groupId);
-    if (segmentGroup == null) return [];
-    let segmentsWithoutPosition: number[] = segmentGroup
+  getSegmentsWithMissingPosition = (groupId: number | null): number[] => {
+    const relevantSegments =
+      groupId != null ? this.getSegmentsOfGroup(groupId) : this.getSelectedSegments();
+    if (relevantSegments == null) return [];
+    let segmentsWithoutPosition: number[] = relevantSegments
       .filter((segment) => segment.somePosition == null)
       .map((segment) => segment.id);
     return segmentsWithoutPosition.sort();
   };
 
-  handlePerSegment(groupId: number, callback: (s: Segment) => void) {
+  getSelectedSegments = (): Segment[] => {
+    const allSegments = this.props.segments;
+    if (allSegments == null) return [];
+    return this.state.selectedIds.segments.map((segmentId) => allSegments.get(segmentId));
+  };
+
+  getSelectedItemKeys = () => {
+    const mappedIdsToKeys = this.state.selectedIds.segments.map(
+      (segmentId) => `segment-${segmentId}`,
+    );
+    if (this.state.selectedIds.group != null) {
+      return mappedIdsToKeys.concat(`group-${this.state.selectedIds.group}`);
+    }
+    return mappedIdsToKeys;
+  };
+
+  getSegmentOrGroupIdsForKeys = (segmentOrGroupKeys: Key[]) => {
+    let selectedIds: { segments: number[]; group: number | null } = { segments: [], group: null };
+    segmentOrGroupKeys.forEach((key) => {
+      const keyAsString = String(key);
+      if (keyAsString.startsWith("group-")) {
+        const regexSplit = keyAsString.split(/-(-?)/);
+        // concatenate id with negative sign for root group. for positive ids, regexSplit[2] is empty
+        const idWithSign = regexSplit[1].concat(regexSplit[2]);
```

Comment:
I think, this can be simplified to:

```suggestion
        // Note that negative ids can be found here, which is why group- is used as a splitter
        const idWithSign = keyAsString.split(/group-/)[1];
```

Reply:
damn, you're right :D